07-regression1.Rmd

```diff
@@ -312,90 +327,99 @@ sacr_tunek_plot <- ggplot(sacr_results, aes(x = neighbors, y = mean)) +
+
+      plots[[i]] <- ggplot(sacr_preds, aes(x = sqft, y = price)) +
+        geom_point(alpha = 0.4) +
+        xlab("House size (square feet)") +
+        ylab("Price (USD)") +
+        scale_y_continuous(labels = dollar_format()) +
+        geom_line(data = sacr_preds, aes(x = sqft, y = .pred), color = "blue") +
+        ggtitle(paste0("K = ", gridvals[[i]]))
+  } else {
+      plots[[i]] <- ggplot(sacr_preds, aes(x = sqft, y = price)) +
+        geom_point(alpha = 0.4) +
+        xlab("House size (square feet)") +
+        ylab("Price (USD)") +
+        scale_y_continuous(labels = dollar_format()) +
+        geom_hline(data = sacr_preds, mapping = aes(x = sqft), yintercept = mean(sacr_preds$price), color = "blue") +
+        ggtitle(paste0("K = ", gridvals[[i]]))
+  }
 }
 grid.arrange(grobs = plots, ncol = 2)
 ```
 
-Based on the plots above, we see that when $K$ = 1, the blue line runs perfectly
-through almost all of our training observations. This happens because our
-predicted values for a given region depend on just a single observation. A
-model like this has high variance and low bias (intuitively, it provides unreliable
-predictions). It has high variance because
-the flexible blue line follows the training observations very closely, and if
-we were to change any one of the training observation data points we would
-change the flexible blue line quite a lot. This means that the blue line
-matches the data we happen to have in this training data set, however, if we
-were to collect another training data set from the Sacramento real estate
-market it likely wouldn't match those observations as well. Another term 
-that we use to collectively describe this phenomenon is *overfitting*.
-
-What about the plot where $K$ is quite large, say $K$ = 450, for example? When
-$K$ = 450 for this data set, the blue line is extremely smooth, and almost
-flat. This happens because our predicted values for a given x value (here home
-size), depend on many many (450) neighbouring observations. A model
-like this has low variance and high bias (intuitively, it provides very reliable,
-but generally very inaccurate predictions). It has low variance because the
-smooth, inflexible blue line does not follow the training observations very
-closely, and if we were to change any one of the training observation data
-points it really wouldn't affect the shape of the smooth blue line at all. This
-means that although the blue line matches does not match the data we happen to
-have in this particular training data set perfectly, if we were to collect
-another training data set from the Sacramento real estate market it likely
-would match those observations equally as well as it matches those in this
-training data set.  Another term that
-we use to collectively describe this kind of model is *underfitting*. 
+Figure \@ref(fig:07-howK) shows that when $K$ = 1, the blue line runs perfectly
+through all of our training observations. 
```

Comment:
This is not quite true - it is false when we have multiple houses of the same size...

Reply:
@ttimbers do you know what tidymodels defaults to if there are multiple at the same value? just include all ties in the average?

Reply:
nvm sorted